### DISCUSSION TITLE
ProHD is not working in Audio Only Mode?

### DISCUSSION THREAD

**Discussion by RocketAlf** [GerryDazoo/Slinger#421] [Q&A]
Reconfigured an unbrick ProHD box  :-)  in an isolated environment. The config.ini file was created in such a way that the box is called up in "Audio Only" mode. Here is the snippet from config.ini (with "Resolution=0"):

[ProHDaudio]
sbtype="Solo/Pro/ProHD"
password=***********
ipaddress=192.168.2.233
;finderid=52D384BE0386DC478C789593BB191157
port=5007
Resolution=0
FrameRate=30
VideoBandwidth=2000
VideoSmoothness=63
VideoSource=1
RemoteLock=no

The connection is terminated within less than a second (Using VLC for playback). Here is the excerpt from the sling.log:

12/06/2022, 00:02:25.7   Streaming request from ('192.168.2.109', 41526)
Slinginfo  "Solo/Pro/ProHD" 0 30 192.168.2.233 5007 3000 10
ProHDaudio Starting Stream for  192.168.2.109:41526
Connecting... ('192.168.2.233', 5007) Control
ProHDaudio Selecting Video Source 1
Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 24, 25, 26, 27, 28, 29, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 46, 48, 62, 68, 70, 72, 74, 75, 76, 77]
VideoParameters: Resolution= 0 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.2.233', 5007) Stream
ProHDaudio Stream started at 12/06/2022, 00:02:29.857  582 0
Traceback (most recent call last):
  File "slingbox_server.py", line 894, in streamer
    msg = process_solo_msg( msg, stream )
  File "slingbox_server.py", line 731, in process_solo_msg
    sn, objoff, objsiz, length = unpack(fmt, msg[off:off+size]);
struct.error: unpack requires a buffer of 17 bytes
ProHDaudio Error Processing Solo Message. Stopping unpack requires a buffer of 17 bytes None
ProHDaudio Shutting down connections
ProHDaudio Logging Out 

Unfortunately, I can't do anything with the error text and therefore don't know how to fix the error. However, the similar configuration with an M1 box works, and playback with video also works.

Can someone help me to solve the problem?

**Comment by GerryDazoo** [comment]
I suspect there is an issue with the code that is expecting a video stream along with the audio. Unfortunately I don't have a ProHD to test with and even if I did I'm not sure I could fix it. That Solo message processing code is a bit of a blackbox to me. I just unraveled some perl code I found on the net and never really understood what it was doing except generating a viewable video stream.
All I can suggest is cranking down the resolution, framerate and video bandwidth to minimize your bit-rate.
Actually, this might be easy to fix. Would you be interested in taking a beta version out for a test drive?
PS. next time please post all the output from the server. There's useful information there. Like if your running the exe or python.
If your running python I can give you a 1 line patch.

**Reply by RocketAlf** [reply]
Thank you very much for the quick reply. Yes, I forgot to mention that I am using "slingbox_server.py" on the RaspberryPi (installed according to the instructions in "README.md"). If you have a beta version or patch to solve my problem, I would appreciate it if you could provide it to me.

In the attached file you will find entries in "sling.log" after entering "sudo systemctl start sling.service" for the following two cases:
1. the IP addresses of the SBs have been commented out
2. the IP addresses of the SBs are effective
[service_start.log](https://github.com/GerryDazoo/Slinger/files/10172029/service_start.log)

And this is the complete config.ini
[config.ini.txt](https://github.com/GerryDazoo/Slinger/files/10172047/config.ini.txt)

Thanks for your efforts

Btw. You mentioned a Perl code for the Slingboxes. You probably mean "kmttg" (?). I've been using "kmttg_v2.4p" on my PC for a long time to watch the SBs locally. Before November 9th I also used it on the go with my mobile phone (Android + termux + perl). That's not possible since November 9th :-(. But it was also a bit cumbersome to display the video using XServer. That's why I'm glad I found slinger. And I want to thank you for your great work on this project.

**Reply by GerryDazoo** [reply]
Yes it was kmttg!
So I have a fix. And it is patchable but first we need to fix the other start up issues you have
"ModuleNotFoundError: No module named 'netifaces'"    Please install the netifaces and Flask modules per the readme. Things will work much better when my code has all the bits it needs.
 
Here's the patch
`Replace line 870         
        if Solo : pksize = 3000

with

        if Solo : 
            if resolution : pksize = 3000
            else: pksize = 1636

**Reply by RocketAlf** [reply]
thank you  ...it is working now :-)

**Comment by GerryDazoo** [comment]
I had a closer look and a simple patch isn't going to work. :( Sorry. I'll continue investigating.

**Reply by sp222222** [reply]
I'm confused, I thought I read a ways back that 3.08 or so with the correct res config.ini setting to zero would work for audio only. which boxes does audio only work on with slinger?

**Reply by GerryDazoo** [reply]
I don't have a ProHD so I couldn't test this. To answer your question audio only works on the newer boxes 350,500,M1,M2
